src/main/java/com/google/api/codegen/CommentPatterns.java

```diff
@@ -0,0 +1,14 @@
+package com.google.api.codegen;
+
+import java.util.regex.Pattern;
+
+/**
+ * Abstract base class for formatting source comments to follow language-idiomatic style.
```

Comment:
In general, I prefer to keep things more modular, and static utility classes are much more modular than abstract base classes.